Tool_use_with_Toolhouse.ipynb

```diff
@@ -0,0 +1,587 @@
+        "from openai import OpenAI\n",
+        "from toolhouse import Toolhouse"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "458ace84",
+      "metadata": {
+        "id": "458ace84"
+      },
+      "source": [
+        "To integrate Together and Toolhouse, you'll need to set up two environment variables: `TOGETHER_API_KEY` and `TOOLHOUSE_API_KEY`. Follow these steps to obtain your API keys:\n",
+        "\n",
+        "* **Together API Key**: Get your key by visiting the [Together Console](https://api.together.ai/).\n",
+        "* **Toolhouse API Key**: Sign up for Toolhouse using [this link](https://join.toolhouse.ai) to receive $150 in credits. You will receive your API key as part of the onboarding step, and you can always, navigate to the [Toolhouse API Keys page](https://app.toolhouse.ai/settings/api-keys) to create and get an API key.\n",
+        "* **Install the X tool**: In your [Toolhouse dashboard](https://app.toolhouse.ai), click Install next to the \"Search X\" tool ([direct link](https://app.toolhouse.ai/store/search_x)).\n",
+        "* **Install the Image Generation tool**: In your [Toolhouse dashboard](https://app.toolhouse.ai), click Install next to the \"Image Generation\" tool ([direct link](https://app.toolhouse.ai/store/image_generation_flux)).\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Set up keys in Colab\n",
+        "\n",
+        "Once you have signe-up and got both API keys, set them as environment variables to start using Together with Toolhouse.\n",
+        "\n",
+        "To do that click on the Key icon in the left menu on Colab and click **\"Add new secret\".**\n",
+        "\n",
+        "Create 2 keys and make sure their **name** is spelled like below:\n",
+        "```\n",
+        "TOGETHER_API_KEY\n",
+        "TOOLHOUSE_API_KEY\n",
+        "```\n",
+        "\n",
+        "Enter the values for each key and switch on the **allow note book access\"**"
+      ],
+      "metadata": {
+        "id": "dEPpWhup3Fva"
+      },
+      "id": "dEPpWhup3Fva"
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# 2. Setting up the SDKs\n",
+        "\n",
+        "Since we've imported the SDKs already, let's initialize them and authenticate ourselves using the API keys we just set.\n",
+        "\n",
+        "If you want to check again, your API keys should be now in the left sidebar of Google Colab after clicking the key icon."
+      ],
+      "metadata": {
+        "id": "_kblVg3pXlMy"
+      },
+      "id": "_kblVg3pXlMy"
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "bf32deff",
+      "metadata": {
+        "id": "bf32deff"
+      },
+      "outputs": [],
+      "source": [
+        "from google.colab import userdata\n",
+        "client = OpenAI(base_url = \"https://api.together.xyz/v1\", api_key=userdata.get(\"TOGETHER_API_KEY\"))\n",
```

Comment:
Can we use the together python API here.

Details in docs here - https://docs.together.ai/docs/quickstart#3-run-your-first-query-against-a-model